src/GameLibs/Snake/System/AppleSystem.cpp

```diff
@@ -25,23 +28,42 @@ Snake::System::AppleSystem::AppleSystem()
 // TODO check the pos of snake for the apple because we don't want
 // the apple to spawn on the snake
 
-void modifyApplePos(Arcade::ECS::IEntityManager &currentEntityManager, std::vector<Arcade::Vector3f> positions)
+void modifyApplePos(Arcade::ECS::IEventManager &eventManager, Arcade::ECS::IEntityManager &currentEntityManager, std::vector<Arcade::Vector3f> positions)
 {
     std::shared_ptr<Arcade::ECS::IEntity> apple = currentEntityManager.getEntitiesById("Apple");
     Arcade::ECS::IComponent &appleIComp = apple->getComponents("apple");
+    auto snakeEntities = currentEntityManager.getEntitiesByComponentType(Arcade::ECS::CompType::MOVEABLE);
+    std::vector<Arcade::Vector3f> tempPositions = positions;
+    std::size_t nbrOfPos = 20;
+    std::size_t randNumber = 1 + (std::rand() % nbrOfPos);
+    Arcade::Vector3f randPosition = positions[randNumber];
 
+    for (auto it = snakeEntities->begin(); it != snakeEntities->end(); it++) {
+        auto entity = *it;
+        auto components = *(entity->getComponents(Arcade::ECS::CompType::SPRITE).begin());
+        Arcade::Graph::Sprite &moveableSprite = static_cast<Arcade::Graph::Sprite &>(*components);
```

Comment:
dereference of reference cast ?

why not shared direct ?

Reply:
Car components est un IComponent sauf que moi je veux le cast en sprite pour accéder à son attribut de position

Reply:
Et le static cast sur un shared le compilo il aime pas trop

Reply:
static_pointer_cast
```suggestion
        std::shared_ptr<Arcade::Graph::Sprite> moveableSprite = static_pointer_cast<Arcade::Graph::Sprite>(components);
```
et si tu veux:
```cpp
auto moveableSprite = static_pointer_cast<Arcade::Graph::Sprite>(components);
```